include/picongpu/particles/particleToGrid/derivedAttributes/MomentumDensity.def

```diff
@@ -75,7 +77,8 @@ namespace picongpu
                     //! Get text name
                     HINLINE static std::string getName()
                     {
-                        return "particleMomentumDensity";
+                        auto const componentNames = plugins::misc::getComponentNames(3);
+                        return "momentumDensity/" + componentNames[T_direction];
```

Comment:
If I am not mistaken, the copy of the string from the vector in the return cannot be elided because the value is accessed by (const) reference from an object, but we can avoid copying the payload by moving.
```suggestion
                        auto componentNames = plugins::misc::getComponentNames(3);
                        return "momentumDensity/" + std::move(componentNames[T_direction]);
```

Reply:
There is no reference in the code because we take the name by copy. `auto const componentNames = ...`

Reply:
`componentNames[T_direction]` is a const reference to an element of the vector `componentNames`.

Reply:
The suggested change is intended to make the compiler use an rvalue overload of `operator+` ([here, overload (11)](https://en.cppreference.com/w/cpp/string/basic_string/operator%2B)).